### PR TITLE
Refactor operator in preparation for leader-election

### LINKED ISSUE #8: Add leader election to transflect
Implement leader election to transflect

### PR DESCRIPTION
**Pull request by juliaogris**
Refactor, reformat, rename and reorganise markdown, yaml and go code
files in preparation for leader election. This PR was created as all
non-functional changes for the leader-election implementation so that
leader-election is purely functional and can be more easily reviewed
with care (see #7 ).

Issue: #8

### CODE REVIEW

File: cmd/transflect-operator/main.go [hunk 1]
```diff
@@ -25,6 +25,7 @@ type config struct {
 	UseIngress bool             `short:"i" help:"Create and use temporary ingress to access temporary service, e.g. from outside cluster"`
 	Address    string           `short:"a" help:"gRPC reflection server address, host:port"`
 	Plaintext  bool             `short:"p" help:"Use plain-text; no TLS"`
+	Namespace  string           `help:"Use custom namespace" default:"transflect"`
```
Comment (camh-) [inline cmd/transflect-operator/main.go]: I'd be more explicit here:
```
LeaseNamespace string `help:"Namespace in which leader election lease is created" default:"transflect"`
```

File: deployment/transflect.yaml [hunk 2]
```diff
@@ -42,50 +42,18 @@ kind: ClusterRole
...
-    resources:
-      - envoyfilters
-    verbs:
-      - create
-      - delete
-      - get
-      - list
-      - update
+  - apiGroups: [ apps ]
+    resources: [ replicasets, deployment ]
```
Comment (camh-) [inline deployment/transflect.yaml]: nit: i guess it doesn't need to be plural, but may as well make it "deployments" for consistency.

**Review by camh- — COMMENTED** [review]
one teeny tiny thing.

File: deployment/transflect.yaml (hunk 2 above)
Comment (juliaogris) [reply deployment/transflect.yaml]: done, thanks.

**Review by jvmakine — APPROVED** [review]

File: cmd/transflect-operator/operator.go [hunk 3]
```diff
@@ -96,20 +81,36 @@ func watchErrorHandler(_ *cache.Reflector, err error) {
...
 	o.deployInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
 		DeleteFunc: o.cleanupDeployment,
 	})
+	log.Debug().Msg("Start leading: informer event handlers registered")
 
-	log.Debug().Msg("Starting: informer event handlers registered")
+	// Run workers and informers
 	o.runWorkers(42)
 	o.rsInformer.Informer().Run(o.stopper)
 	o.deployInformer.Informer().Run(o.stopper)
```
Comment (camh-) [inline cmd/transflect-operator/operator.go]: These two calls to `Run()` are not done asynchronously. From the Run docs:

> Run starts and runs the shared informer, returning after it stops.

So the deployInformer does not start until the rsInformer stops.

Comment (juliaogris) [reply cmd/transflect-operator/operator.go]: yesss!! thanks for the eagle eye yet again.

**Review by camh- — APPROVED** [review]
LGTM :green_apple:

**Pull request merged**